Core/JsRuntime/Include/Babylon/JsConsoleLogger.h

```diff
@@ -0,0 +1,22 @@
+#pragma once
```

Comment:
I'm not sure this helper class belongs in JsRuntime, but I also don't have a better idea. Maybe we can keep it in NativeEngine folder for now? @syntheticmagus Any thoughts?

Reply:
_Hmm_, funky. It feels like it wants to be part of the `Console` Polyfill, but other native Components can't take dependencies on Polyfills.

Maybe the console functionality needs to be split in two, with the core functionality housed in a Plugin (or even Core) that other native Components can depend on, and the `Console` Polyfill that exposes the functionality to JS depends on that. Thoughts?

Reply:
Some JS engine exposes a Console object by default (like JSC) and if we are integrating with an existing environment, it may already have Console (like React Native). I don't think it makes sense to split the functionality.

Reply:
In that case, is it worth encapsulating this at all? It looks like both the places it's currently being called from have access to `Napi::CallbackInfo`s, and I would expect anywhere we'd want to log to JS from should at least have _some_ way to get a `Napi::Env`. `info.Env().Global().Get("console").As<Napi::Object>().Get("log").As<Napi::Function>().Call(...)` isn't exactly concise, but it's not bespoke, requires no infrastructure, directly hooks into whatever the JS has decided to do for the console, and will only matter speed-wise if we're spewing logs, which will be expensive anyway.

Reply:
I guess another option if we wanted encapsulate to avoid the verbosity would be to add something to `napi-extensions` alongside napi_pointer.h, maybe napi_console.h or something. Then we could hide the above console-getting in helper functions like `Napi::Console::Log(Napi::Env, const char*)`, `Napi::Console::Warn(Napi::Env*, const char*)`, etc.

Reply:
I don't like the napi_console.h in napi itself. Maybe a utility dependency outside of napi would be fine. This is effectively a set of helper functions. I think maybe caching the `console.*` functions in NativeEngine as members isn't too bad. Basically, any code that wants to log to the console can do something similar. @SergioRZMasson any thoughts?

Reply:
I think this is too little of a utility to add to napi, it was designed just to avoid typing the entire thing every time and to avoid other people that might use the log to forget to check if console exist. Performance wise I think this can be considered a special case, a heathy application should not be throwing's warnings at the console anyway, I didn't want to start putting extra stuff on NativeEngine's members, unless they are essential.

Reply:
I think a simple helper function into a utility header(maybe inside NativeEngine project itself?) can be considered fine?

Reply:
I don't have a preference. I was thinking you can move these files into NativeEngine in my original reply. That would be fine with me.